src/js/components/ResponsiveContext.tsx

```diff
@@ -0,0 +1,49 @@
+import type { ReactNode } from 'react';
+import React, { createContext, useContext, useEffect, useState } from 'react';
+
+import { DeviceBreakpoints } from '@/constants/deviceBreakpoints';
+
+interface ResponsiveContextType {
+  isMobile: boolean;
+  isTablet: boolean;
+}
+
+const DefaultResponsiveContext: ResponsiveContextType = {
+  isMobile: false,
+  isTablet: false,
+};
+
+const ResponsiveContext = createContext<ResponsiveContextType>(DefaultResponsiveContext);
+
+interface ResponsiveProviderProps {
+  children: ReactNode;
+}
+
+export const ResponsiveProvider = ({ children }: ResponsiveProviderProps) => {
+  const [isMobile, setIsMobile] = useState<boolean>(false);
+  const [isTablet, setIsTablet] = useState<boolean>(false);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobile(window.innerWidth <= DeviceBreakpoints.MOBILE);
+      setIsTablet(window.innerWidth > DeviceBreakpoints.MOBILE && window.innerWidth <= DeviceBreakpoints.TABLET);
+    };
+    handleResize();
```

Comment:
instead of this, set initial state as above - factor out above set logic into small functions for isMobile and isTablet